vscode-ext/src/tests/e2e.ts

```diff
@@ -0,0 +1,146 @@
+import { existsSync } from "fs";
+import { expect } from "chai";
+import { join } from "path";
+import { VSBrowser, StatusBar, TextEditor, EditorView, ActivityBar, DebugView, InputBox, DebugToolbar } from "vscode-extension-tester";
+import get from "axios";
+
+
+// This suite tests basic flow of mirroring traffic from remote pod
+// - Enable mirrord -> Disable mirrord
+// - Create mirrord config by pressing the gear icon
+// - Set a breakpoint in the python file
+// - Start debugging the python file
+// - Select the pod from the QuickPick
+// - Send traffic to the pod
+// - Tests successfully exit if breakpoint is hit
+const kubeService = process.env.KUBE_SERVICE;
+const podToSelect = process.env.POD_TO_SELECT;
+
+describe("mirrord sample flow test", function () {
+
+    this.timeout(1000000); // --> mocha tests timeout
+    this.bail(true); // --> stop tests on first failure
+
+    let browser: VSBrowser;
+    let debugToolbar: DebugToolbar;
+
+    const testWorkspace = join(__dirname, '../../test-workspace');
+    const fileName = "app_flask.py";
+    const mirrordConfigPath = join(testWorkspace, '.mirrord/mirrord.json');
+    const configurationFile = "Python: Current File";
+    const breakPoint = 9;
+    const DEFAULT_CONDITION_TIMEOUT = 10000;
+
+    before(async function () {
+        console.log("podToSelect: " + podToSelect);
+        console.log("kubeService: " + kubeService);
+
+        expect(podToSelect).to.not.be.undefined;
+        expect(kubeService).to.not.be.undefined;
+
+        browser = VSBrowser.instance;
+        // need to bring the flask app in open editors
+        await browser.openResources(testWorkspace, join(testWorkspace, fileName));
+    });
+
+    after(async function () {
+        if (await debugToolbar?.isDisplayed()) {
+            await debugToolbar.stop();
+        }
+    });
+
+    it("enable mirrord", async function () {
+        const statusBar = new StatusBar();
+        await browser.driver.wait(async () => {
+            const enableButton = await statusBar.getItem("Enable mirrord");
+            if (enableButton !== undefined) {
```

Comment:
In which cases would this be undefined?

Reply:
its undefined when it has not loaded

Reply:
Is that feasible? If so, shouldn't there be an else clause? If not, shouldn't it just be enableButton?.click();?

Reply:
it automatically means the condition is not true if we don't return true

Reply:
wait requires a boolean